forwarder.go

```diff
@@ -10,24 +10,44 @@ import (
 	"net/http"
 	"os"
 	"strings"
+	"sync"
 )
 
+const workers = 8
+
 func main() {
 	log.Println("Splunk forwarder: Started")
 	defer log.Println("Splunk forwarder: Stopped")
 
+	forSplunk := make(chan string)
+
+	var wg sync.WaitGroup
+
+	for i := 0; i < workers; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for msg := range forSplunk {
+				postToSplunk(msg)					
+			}
+		}()
+	}
+
 	br := bufio.NewReader(os.Stdin)
 	for {
 		str, err := br.ReadString('\n')
 
 		if err != nil {
 			if err == io.EOF {
+				close(forSplunk)
 				return
 			}
 			log.Fatal(err)
 		}
-		postToSplunk(str)
+		forSplunk <- str
 	}
+
+	wg.Wait()
```

Comment:
@khatton-ft comment is related to the following block that never exits.

	for {
		str, err := br.ReadString('\n')

		if err != nil {
			if err == io.EOF {
				close(forSplunk)
				return
			}
			log.Fatal(err)
		}
		forSplunk <- str
	}

The 	wg.Wait() below the block therefore never gets called.